services/src/main/java/org/keycloak/services/resources/WelcomeResource.java

```diff
@@ -176,6 +179,7 @@ private Response createWelcomePage(String successMessage, String errorMessage) {
 
             Map<String, Object> map = new HashMap<>();
 
+            map.put("adminConsoleEnabled", ADMIN_CONSOLE_ENABLED);
```

Comment:
It should be fine to use `Profile.isFeatureEnabled(Profile.Feature.ADMIN2) || Profile.isFeatureEnabled(Profile.Feature.ADMIN)` directly here.

Quarkus applications have a well-defined initialization process and using `Profile` in static init might cause issues as they are executed most of the time when the class is loaded.

Reply:
Good point I'll adapt the code